default/python/AI/buildings.py

```diff
@@ -39,22 +74,29 @@ def enqueue(self, pid: PlanetId) -> bool:
 
     def available(self) -> bool:
         """
-        Return true if this building is available for empire.
+        Return true if this building is available for empire and may be built by this AI.
         """
-        return fo.getEmpire().buildingTypeAvailable(self.value)
+        character = get_aistate().character
+        return character.may_build_building(self.value) and fo.getEmpire().buildingTypeAvailable(self.value)
 
     def get_opinions(self) -> Opinion:
         """
         Returns opinions about the building at the empire's planets.
         """
         return get_planet_opinion(self.value)
 
-    def queued_in(self) -> List[PlanetId]:
+    def queued_at(self) -> List[PlanetId]:
```

Comment:
This change produces a lot of noise to PR.  Please create a separate one with it. 

These kinds of changes have a high chance to be merged without comment on the same day.